apps/src/storage/dataBrowser/dataVisualizer/Snapshot.jsx

```diff
@@ -0,0 +1,122 @@
+/* global ClipboardItem */
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import moment from 'moment';
+import msg from '@cdo/locale';
+import {html2canvas} from '@cdo/apps/util/htmlToCanvasWrapper';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import PendingButton from '@cdo/apps/templates/PendingButton';
+import * as dataStyles from '../dataStyles';
+
+class Snapshot extends React.Component {
+  static propTypes = {
+    chartTitle: PropTypes.string.isRequired,
+    selectedOptions: PropTypes.string.isRequired,
+    // Provided via Redux
+    tableName: PropTypes.string.isRequired,
+    projectName: PropTypes.string.isRequired
+  };
+
+  state = {
+    isSnapshotOpen: false,
+    isCopyPending: false,
+    isSavePending: false
+  };
+
+  handleOpen = () => this.setState({isSnapshotOpen: true});
+  handleClose = () => this.setState({isSnapshotOpen: false});
+
+  getPngBlob = () => {
+    const element = document.getElementById('snapshot');
+    const options = {
+      background: '#fff'
+    };
+    return html2canvas(element, options).then(canvas => {
+      return new Promise(function(resolve, reject) {
+        canvas.toBlob(function(blob) {
+          resolve(blob);
+        });
+      });
+    });
+  };
+
+  copy = () => {
+    this.setState({isCopyPending: true});
+    this.getPngBlob().then(pngBlob => {
+      try {
+        navigator.clipboard.write([
+          new ClipboardItem({
+            'image/png': pngBlob
+          })
+        ]);
+        this.setState({isCopyPending: false});
+      } catch (error) {
+        console.error(error);
+      }
+    });
+  };
+
+  save = () => {
+    this.setState({isSavePending: true});
+    this.getPngBlob().then(pngBlob => {
+      const download = document.createElement('a');
+      download.href = URL.createObjectURL(pngBlob);
+      download.download = 'image.png';
+      download.click();
+      this.setState({isSavePending: false});
+    });
+  };
+
+  render() {
+    return (
+      <div>
+        <button
+          type="button"
+          style={dataStyles.grayButton}
+          onClick={this.handleOpen}
+        >
+          {msg.dataVisualizerViewSnapshot()}
+        </button>
+        <BaseDialog
+          isOpen={this.state.isSnapshotOpen}
+          handleClose={this.handleClose}
+          fullWidth
+          fullHeight
+        >
+          <div id="snapshot">
+            <h1>{this.props.chartTitle}</h1>
+            <img src={require('./placeholder.png')} />
```

Comment:
i'm pretty sure this `require('./placeholder.png')` is temporary, but if it's not, let's move the import statement to the top of the file for clarity. (although we may want to end up using this placeholder while we're loading the actual chart image? in which case it won't be going away)